core/lib/snapshots_applier/src/lib.rs

```diff
@@ -685,9 +691,9 @@ impl<'a> SnapshotsApplier<'a> {
         .await?;
         tracing::info!("Chosen snapshot recovery strategy: {strategy:?} with status: {applied_snapshot_status:?}");
         let (created_from_scratch, snapshot_version) = match strategy {
-            SnapshotRecoveryStrategy::Completed => return Ok((strategy, applied_snapshot_status)),
             SnapshotRecoveryStrategy::New(version) => (true, version),
             SnapshotRecoveryStrategy::Resumed(version) => (false, version),
+            SnapshotRecoveryStrategy::Completed => (false, SnapshotVersion::Version1),
```

Comment:
Did you mean to hard code `Version1` here instead of `applied_snapshot_status`? Feels like a potential pitfall if we ever introduce a new version.

Reply:
Refactored this place to avoid relying on such hacks.